compiler/types.nim

```diff
@@ -1172,7 +1172,7 @@ proc typeAllowedAux(marker: var IntSet, typ: PType, kind: TSymKind,
   result = nil
   if typ == nil: return
   if containsOrIncl(marker, typ.id): return
-  var t = skipTypes(typ, abstractInst-{tyTypeDesc})
+  var t = skipTypes(typ, abstractInst+{tyStatic}-{tyTypeDesc})
```

Comment:
This change is unrelated to your PR.

Reply:
Why did it work before? The compiler code is quite clear, you cannot put a static into a tuple.

Reply:
> Why did it work before? 

because support for it had been added here: https://github.com/nim-lang/Nim/pull/9046, but that logic failed after current PR added ` `compiles(...)` inside `proc `$`*[T: tuple|object](x: T): string =`

I pushed a commit that reverts the change in compiler/types.nim and modifies ./tests/vm/tissues.nim to not rely on having a static in a tuple, marking as resolved.